chronos/asyncfutures2.nim

```diff
@@ -398,6 +402,49 @@ proc `cancelCallback=`*[T](future: Future[T], cb: CallbackFunc) =
   ## This callback will be called immediately as ``future.cancel()`` invoked.
   future.cancelcb = cb
 
+{.push stackTrace: off.}
+proc internalContinue[T](fut: pointer) {.gcsafe, raises: [Defect].}
+
+proc futureContinue*[T](fut: Future[T]) {.gcsafe, raises: [Defect].} =
+
+  # Used internally by async transformation
+  try:
+    if not(fut.closure.finished()):
+      var next = fut.closure(fut)
+      # Continue while the yielded future is already finished.
+      while (not next.isNil()) and next.finished():
+        next = fut.closure(fut)
+        if fut.closure.finished():
+          break
+
+      if next == nil:
+        if not(fut.finished()):
+          # Can't get strName anymore
```

Comment:
you can, by passing it as a `cstring` parameter - the question is whether it's worth it

Reply:
Actually we could show the future location in this message, we already have it (just though about it)